s3transfer/utils.py

```diff
@@ -296,7 +298,8 @@ def __init__(self, filename, start_byte=0, mode='rb', open_function=open):
     def _open_if_needed(self):
         if self._fileobj is None:
             self._fileobj = self._open_function(self._filename, self._mode)
-            self._fileobj.seek(self._start_byte)
+            if self._start_byte != 0:
```

Comment:
I think we should add a test to make sure we do not seek anymore when start byte is 0.